Automattic/woocommerce-payments
# Run wp-prettier in pre-commit hook

This PR adds https://github.com/Automattic/wp-prettier to our dev dependencies and formats our files during commit hooks. The configuration is ripped straight from wc-admin.

I thought this would be a pretty interesting experiment, and might be a nice way to keep us honest about sticking to code standards.

I needed to make a couple of changes to our dev dependencies for this to work properly. As noted in the discussions below there were some conflicts with ESLint, and the same happened with Stylelint.

In order to fix these conflicts I had to

1. Update Stylelint
1. Install some eslint and stylelint configs that disable linting rules that conflict with prettier
1. Add special rules in the eslint and stylelint configurations to fit the WP standard, as opposed to the traditional JS/CSS standards.

Because of the updated configuration for the CSS linter I actually had to make a couple of minor changes to some CSS files, but those changes can pretty much be ignored during the review.

#### Changes proposed in this Pull Request

* Run wp-prettier on commit.
* Adds CSS and JS formatting commands via wp-prettier:
    * `npm run format` - formats **all** JS and CSS files in the project.
    * `npm run format:js` - formats **all** JS files in the project.
    * `npm run format:css` - formats **all** CSS files in the project.
    * `npm run format:provided <file_1> <file_2> '<glob>'` - formats the provided files and/or globs.

#### Testing instructions



* Make some changes that don't follow our coding standards
* Commit changes
* Observe that the file has been formatted

OR

* Make some changes that don't follow our coding standards
* run `npm run format:provided <changed_file>`
* Observe that the file has been formatted

-------------------
